app/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/util/PreValidateTraderETDSHttpReader.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.connectors.util
+
+import play.api.Logging
+import play.api.http.Status.{BAD_REQUEST, NOT_FOUND, UNAUTHORIZED}
+import play.api.libs.json.Json
+import play.api.mvc.Result
+import play.api.mvc.Results.Status
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.EisErrorResponsePresentation
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.eis.EISErrorMessage
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.preValidateTrader.response.ExciseTraderValidationETDSResponse
+import uk.gov.hmrc.excisemovementcontrolsystemapi.utils.DateTimeService
+import uk.gov.hmrc.http.HttpErrorFunctions.is2xx
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+
+class PreValidateTraderETDSHttpReader(
```

Comment:
Will the non-ETDS HTTP reader be removed when we go live with EDTS? If so, can we rename this at that point.